OpenRA.Mods.Common/Traits/SpawnActorOnDeath.cs

```diff
@@ -98,8 +109,24 @@ void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
 				new FactionInit(faction)
 			};
 
+			if (self.EffectiveOwner != null && self.EffectiveOwner.Disguised)
+				td.Add(new EffectiveOwnerInit(self.EffectiveOwner.Owner));
+			else if (Info.EffectiveOwnerFromOwner)
+				td.Add(new EffectiveOwnerInit(self.Owner));
+
 			if (Info.OwnerType == OwnerType.Victim)
-				td.Add(new OwnerInit(self.Owner));
+			{
+				// Fall back to InternalOwner if the Victim was defeated,
+				// but only if InternalOwner is defined
+				if (!defeated || string.IsNullOrEmpty(Info.InternalOwner))
```

Comment:
This sounds like able to recreating the bug the ticket is about if I just empty out InternalOwner with SpawnAfterDefeat: true.

Reply:
It is intentional then (if you put those values), so not really an issue? (Someone might want to do it after all.)

Reply:
Meh, alright - https://github.com/OpenRA/OpenRA/pull/9117#discussion_r37540853 logic can be applied to this case as well.